core/hoverfly_test.go

```diff
@@ -554,6 +553,104 @@ func Test_Hoverfly_Save_SavesRequestAndResponseToSimulation(t *testing.T) {
 	Expect(unit.Simulation.Templates[0].Response.Status).To(Equal(200))
 }
 
+func Test_Hoverfly_Save_DoesNotSaveRequestHeadersWhenGivenHeadersArrayIsNil(t *testing.T) {
+	RegisterTestingT(t)
+
+	unit := NewHoverflyWithConfiguration(&Configuration{})
+
+	unit.Save(&models.RequestDetails{
+		Headers: map[string][]string{"testheader": []string{"testvalue"}},
+	}, &models.ResponseDetails{
+		Body:    "testresponsebody",
+		Headers: map[string][]string{"testheader": []string{"testvalue"}},
+		Status:  200,
+	}, nil)
```

Comment:
It feels like this should be the other way round with the use of `nil` and empty `slice`. See my previous comment.